src/audio/switch.c

```diff
@@ -11,10 +11,25 @@
 #include <user/trace.h>
 #include <stddef.h>
 
-/* tracing */
-#define trace_switch(__e) trace_event(TRACE_CLASS_SWITCH, __e)
-#define trace_switch_error(__e)   trace_error(TRACE_CLASS_SWITCH, __e)
-#define tracev_switch(__e)        tracev_event(TRACE_CLASS_SWITCH, __e)
+/* mixer tracing */
+#define trace_switch(__e, ...)					\
+	trace_event(TRACE_CLASS_SWITCH, __e, ##__VA_ARGS__)
+#define trace_switch_with_ids(comp_ptr, __e, ...) do {		\
```

Comment:
I think these lone `do{`s shouldn't be here

Reply:
Thats righ